packages/react-ui/src/widgets/PieWidgetUI/PieWidgetUI.js

```diff
@@ -98,38 +80,52 @@ function PieWidgetUI({
     [formatter, theme.palette.common.white, theme.palette.black, tooltipFormatter]
   );
 
-  // Series
-  const labelOptions = useMemo(
-    () => ({
-      formatter: '{per|{d}%}\n{b|{b}}',
-      position: 'center',
-      rich: {
-        b: {
-          fontFamily: theme.typography.overlineDelicate.fontFamily,
-          fontSize: theme.spacingValue * 1.75,
-          lineHeight: theme.spacingValue * 1.75,
-          fontWeight: 'normal',
-          color: theme.palette.text.primary
-        },
-        per: {
-          ...theme.typography,
-          fontSize: theme.spacingValue * 3,
-          lineHeight: theme.spacingValue * 4.5,
-          fontWeight: 600,
-          color: theme.palette.text.primary
-        }
-      }
-    }),
-    [theme]
-  );
+  // Select the largest category to display in CentralText and calculate its percentage from the total
+  const topSelectedCategory = useMemo(() => {
+    if (!processedData || processedData.length === 0) {
+      return null;
+    }
+
+    let array;
+    if (selectedCategories.length > 0) {
+      array = processedData.filter((dataItem) =>
+        selectedCategories.includes(dataItem.name)
+      );
+    } else {
+      array = processedData;
+    }
+
+    const largestCategory = findLargestCategory(array);
+    const category = array.find((element) => element === largestCategory);
+
+    if (!category) {
+      return null;
+    }
+
+    let sumValue = 0;
+    for (const category of processedData) {
+      sumValue += category.value;
+    }
+
+    const percentage = calculatePercentage(category.value, sumValue);
+    category.percentage = percentage;
+
+    return category;
+  }, [processedData, selectedCategories]);
```

Comment:
Didn't you consider putting this hook outside, as a useTopSelectedCategory?

Reply:
I moved to the `CentralText` component